Create pull_request_template.md (#112)
canonical/ubuntu-server-documentation: .github/pull_request_template.md

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

> ### Description
> Please provide a clear and concise description of your changes. Explain **what** this pull request accomplishes and *

Reviewer: ```suggestion
### Description

Please provide a clear and concise description of your changes. Explain **what** this pull request accomplishes and *
```

Reviewer: (for consistency with the other sections)

*why** it is needed.

---

### Related Issue

> If this pull request addresses an existing issue, please link to it below. Use the `Fixes #<issue-number>` syntax to

Reviewer 2: Thanks for your help, already good - a few suggestions to make it even better.
Fixes could point as a link to https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue for those not aware of the details.

> close the issue automatically when the pull request is merged.

Reviewer: ```suggestion
[close the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue) automatically when the pull request is merged.
```

Example:

- Fixes: #1234

---

### Contributor License Agreement (CLA)

By contributing to this project, you agree to the terms of
the [Canonical Contributor License Agreement (CLA)](https://ubuntu.com/legal/contributors).  

> If you have not already signed the CLA, please do so [here](https://ubuntu.com/legal/contributors).

Reviewer: ```suggestion
If you have not already signed the CLA, [please do so here](https://ubuntu.com/legal/contributors).
```

Reviewer: For accessibility, it's better to include a slightly longer text for links so it provides a larger area for people using magnifiers to click on

---

### Commit Message for Squash Merge

> If this pull request is merged with the squash option, what should the commit message be?  

Reviewer 2: Since the submitter can't know if we merge one or the other way that sentence can be misleading.
Some have well balanced commits and are better merged normal, other have many minor commits and are better squashed.

I'd leave it to the PR author to suggest and hence something like: "if you’d want this to be squashed on merge, ..." to lead into this paragraph?

Author: I agree @cpaelzer

> If this pull request is merged with the squash option, what should the commit message be?  

Reviewer: ```suggestion
We typically squash commits when merging. You can specify the commit message that should be used in this case if you wish.
```

Provide the desired commit message below:

Example:

> [Category] Brief description of changes made

Reviewer: Since this repository only contains docs (and not code) it's probably not needed to use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), but if we do use them we should link to something that will help contributors understand what the available categories are. If you've got different categories in mind (since all the changes are docs changes), I think it would be good to list them here.

---

### Checklist

- [ ] I have read and followed the [Ubuntu Server contributing guide](https://documentation.ubuntu.com/server/contributing/).
- [ ] I have signed the [Contributor License Agreement (CLA)](https://ubuntu.com/legal/contributors).

> - [ ] My changes are well-documented, and I have updated the documentation as needed.

Reviewer: I think we probably won't need this one, since all the changes in this repo relate to documentation :)

Reviewer 2: yes meta - contributions here are documentation by definition - so this line could probably go away indeed

- [ ] My pull request is linked to an existing issue (if applicable).
- [ ] I have tested my changes, and they work as expected.

> - [ ] New and existing unit tests pass locally with my changes.

Reviewer 2: Unit tests in contributions to this repository likely never have unit-tests as it is not code.
Some time in the future we most likely re-enable some spell checks and other linting and once we do so we'd add a better matching line here then. For now I'd suggest to remove it.

> - [ ] New and existing unit tests pass locally with my changes.

Reviewer: ```suggestion
```
